openssl/src/ssl/mod.rs

```diff
@@ -847,11 +858,31 @@ impl SslContextBuilder {
         }
     }
 
-    /// Sets the mode used by the context, returning the previous mode.
+    /// Sets the mode used by the context, returning the new mode bit mask.
+    ///
+    /// Options already set before are not cleared.
     #[corresponds(SSL_CTX_set_mode)]
     pub fn set_mode(&mut self, mode: SslMode) -> SslMode {
         unsafe {
-            let bits = ffi::SSL_CTX_set_mode(self.as_ptr(), mode.bits() as MtuTy) as SslBitType;
+            let bits = ffi::SSL_CTX_set_mode(self.as_ptr(), mode.bits() as ModeTy) as SslBitType;
+            SslMode::from_bits_retain(bits)
+        }
+    }
+
+    /// Clear the mode used by the context, returning the new mode bit mask.
```

Comment:
This only clears the set bits of the `mode` argument, right?

Reply:
yes, updated